### PR TITLE
upgrade: upgrade the vendor of cri-o/ocicni

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did

At present, `cri-o/ocicni` will watch CNI configuration file through inotify and quit if inofity runs out without any error report, it's not make sense.  Fortunately, the issue has been settled at https://github.com/cri-o/ocicni/commit/08318295d0b14f3acde0a0431434858324387bee . So we update the vendor of  `cri-o/ocicni`

### Ⅱ. Does this pull request fix one issue?

None.
### Ⅳ. Describe how to verify it
use command `sudo sysctl -w fs.inotify.max_user_instances=10` to limit the inotify, and you will get error like this:
```
# pouchd --enable-cri
INFO[2018-09-28T13:00:41.374491472Z] ctrd: new containerd process, pid: 14674     
INFO[0000] starting containerd                           module=containerd revision=773c489c9c1b21a6d78b5c538cd395416ec50f88 version=v1.0.3
INFO[0000] loading plugin "io.containerd.content.v1.content"...  module=containerd type=io.containerd.content.v1
INFO[0000] loading plugin "io.containerd.snapshotter.v1.btrfs"...  module=containerd type=io.containerd.snapshotter.v1
WARN[0000] failed to load plugin io.containerd.snapshotter.v1.btrfs  error="path /var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter" module=containerd
INFO[0000] loading plugin "io.containerd.snapshotter.v1.overlayfs"...  module=containerd type=io.containerd.snapshotter.v1
INFO[0000] loading plugin "io.containerd.metadata.v1.bolt"...  module=containerd type=io.containerd.metadata.v1
WARN[0000] could not use snapshotter btrfs in metadata plugin  error="path /var/lib/pouch/containerd/root/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter" module="containerd/io.containerd.metadata.v1.bolt"
INFO[0000] loading plugin "io.containerd.differ.v1.walking"...  module=containerd type=io.containerd.differ.v1
INFO[0000] loading plugin "io.containerd.gc.v1.scheduler"...  module=containerd type=io.containerd.gc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.containers"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.content"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.diff"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.events"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.healthcheck"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.images"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.leases"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.namespaces"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.snapshots"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.monitor.v1.cgroups"...  module=containerd type=io.containerd.monitor.v1
INFO[0000] loading plugin "io.containerd.runtime.v1.linux"...  module=containerd type=io.containerd.runtime.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.tasks"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.version"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] loading plugin "io.containerd.grpc.v1.introspection"...  module=containerd type=io.containerd.grpc.v1
INFO[0000] serving...                                    address="/run/containerd/debug.sock" module="containerd/debug"
INFO[0000] serving...                                    address="/var/run/containerd.sock" module="containerd/grpc"
INFO[0000] containerd successfully booted in 0.003135s   module=containerd
INFO[2018-09-28T13:00:41.399856096Z] success to create 5 containerd clients, connect to: /var/run/containerd.sock 
INFO[2018-09-28T13:00:41.537924242Z] Start CRI service with CRI version: v1alpha2 
INFO[2018-09-28T13:00:41.538337026Z] Stream Server will bind to address 10.140.0.7:10010 
INFO[2018-09-28T13:00:41.538553827Z] start new CNI                                
INFO[2018-09-28T13:00:41.538605687Z] start init CNI                               
INFO[2018-09-28T13:00:41.53884408Z] init CNI success                             
WARN[2018-09-28T13:00:41.541473527Z] failed to find group pouch, cannot change unix socket /var/run/pouchd.sock to pouch group 
INFO[2018-09-28T13:00:41.541559456Z] start to listen to: unix:///var/run/pouchd.sock 
Error: failed to get CriManager with error: failed to create cni manager: could not create new watcher too many open files
ERRO[2018-09-28T13:00:41.541667407Z] failed to get CriManager with error: failed to create cni manager: could not create new watcher too many open files 

``` 

### Ⅴ. Special notes for reviews

BTY, update the vendor of `github.com/containernetworking/cni`.

### CODE REVIEW

File: cri/ocicni/cni_manager.go [hunk 1]
```diff
@@ -87,12 +88,22 @@ func (c *CniManager) GetPodNetworkStatus(netnsPath string) (string, error) {
...
+	var err error
+	results, err := c.plugin.GetPodNetworkStatus(podNetwork)
 	if err != nil {
 		return "", fmt.Errorf("failed to get pod network status: %v", err)
 	}
 
-	return ip, nil
+	if len(results) > 0 {
+		// NOTE: only get the first IP of the first network at present
+		result := results[0].(*cnicurrent.Result)
```
Comment (fuweid) [inline cri/ocicni/cni_manager.go]: could we use (val, ok) to do assertion? because it's safe 😅

**Comment by allencloud** [comment]
Any update on this? Still WIP? @Starnop

**Comment by Starnop** [comment]
@allencloud has been updated.  :)

**Comment by Starnop** [comment]
@fuweid @YaoZengzeng PTAL. ^_^

File: cri/ocicni/cni_manager.go [hunk 2]
```diff
@@ -87,12 +88,25 @@ func (c *CniManager) GetPodNetworkStatus(netnsPath string) (string, error) {
...
+	results, err := c.plugin.GetPodNetworkStatus(podNetwork)
 	if err != nil {
 		return "", fmt.Errorf("failed to get pod network status: %v", err)
 	}
 
-	return ip, nil
+	if len(results) > 0 {
+		// NOTE: only get the first IP of the first network at present
+		// TODO: support multiple network
+		if result, ok := results[0].(*cnicurrent.Result); ok {
```
Comment (fuweid) [inline cri/ocicni/cni_manager.go]: so, is the first one the `*cnicurrent.Result`?

Comment (Starnop) [reply cri/ocicni/cni_manager.go]: Yeah. as comments mentioned, we only get the first IP of the first network at present. :)

File: cri/ocicni/cni_manager.go [hunk 3]
```diff
@@ -32,7 +33,7 @@ func NewCniManager(cfg *config.Config) (CniMgr, error) {
 		}
 	}
 
-	plugin, err := ocicni.InitCNI(networkPluginConfDir, networkPluginBinDir)
+	plugin, err := ocicni.InitCNI("", networkPluginConfDir, networkPluginBinDir)
```
Comment (fuweid) [inline cri/ocicni/cni_manager.go]: @Starnop could you comment here for the reason why we set it empty?

File: cri/ocicni/cni_manager.go [hunk 4]
```diff
@@ -87,12 +88,25 @@ func (c *CniManager) GetPodNetworkStatus(netnsPath string) (string, error) {
...
 
-	ip, err := c.plugin.GetPodNetworkStatus(podNetwork)
+	var err error
+	results, err := c.plugin.GetPodNetworkStatus(podNetwork)
 	if err != nil {
 		return "", fmt.Errorf("failed to get pod network status: %v", err)
 	}
 
-	return ip, nil
+	if len(results) > 0 {
```
Comment (allencloud) [inline cri/ocicni/cni_manager.go]: I suggest that we could use `return fast` to ident less.

```
if len(results)==0{
    return "", fmt.Errorf("failed to get pod network status for nil result")
}

if result, ok := results[0].(*cnicurrent.Result); !ok {
    return "", fmt.Errorf("failed to get pod network status for wrong result: %+v", results[0])
}

if len(result.IPs) == 0 {
    return "", fmt.Errorf("failed to get pod network status for nil IP")
}

ip := result.IPs[0].Address.IP.String()
return ip, nil
```

**Comment by allencloud** [comment]
I think we could merge this until we finish some minor change. @Starnop

**Comment by Starnop** [comment]
All above mentioned has been updated. :)

**Comment by allencloud** [comment]
LGTM by @fuweid

**Pull request merged**